### ISSUE TITLE
JSON_EXT Put Null Value

### ISSUE THREAD

**Issue by rodriformiga** [pljson/pljson#34]
I can't put a NULL value using **json_ext** package, but i can with **json** package.
```
--Example
DECLARE
   obj   JSON   := JSON();
   v_num NUMBER := NULL;
BEGIN
   obj.put('X', v_num);           --This is Ok
   json_ext.put(obj, 'z', v_num); --But, this cause a Error
   obj.print();
END;
```

**Comment by jsumners** [comment]
Please provide some more information. An example would be quite helpful.

**Comment by rodriformiga** [comment]
Exemplified ;)

**Comment by jsumners** [comment]
The json_ext module is a high level type safe module. The json module is the core module that allows you to do things that maybe shouldn't be done. In fact, the RFC does not allow for a "number" value to take on null -- http://rfc7159.net/rfc7159#rfc.section.6 . So, if anything, the "bug" is in the opposite direction -- the base json type should not allow a null value (provided that wouldn't break the rest of the package).

https://github.com/pljson/pljson/blob/master/src/json_ext.sql#L489
https://github.com/pljson/pljson/blob/master/src/json_body.typ#L179

**Comment by rodriformiga** [comment]
Ok, but the **json value** may be null under [Section3](http://rfc7159.net/rfc7159#rfc.section.3):
> A JSON value MUST be an **object**, **array**, **number**, or **string**, or one of the following three literal names:
>- false
>- null
>- true

### Example
```
{
  "num1": 111,
  "num2": null,
  "num3": -55
}
```

**Comment by jsumners** [comment]
Yes, that is true. But pl/json is being more strict.

If you think it should behave differently, please submit a patch. But it looks to me like every json_ext `put` method would need to handle the `null` case.

**Comment by boriborm** [comment]
@rodriformiga Try this:

```pl/sql
   json_ext.put(obj, 'z', nvl( v_num, json_value() ) ); 
```

**Comment by jsumners** [comment]
Fixed by pull request #35
